features/admin.server-configurations.v1/forms/username-recovery-form.tsx

```diff
@@ -100,34 +106,102 @@ export const UsernameRecoveryConfigurationForm: FunctionComponent<UsernameRecove
             >
+                <Heading as="h4" className="mb-5">
+                    { t("extensions:manage.serverConfigurations.accountRecovery." +
+                            "passwordRecovery.recoveryOptionHeading") as ReactNode }
+                </Heading>
+
+                <Field.Checkbox
+                    className="mb-5"
+                    ariaLabel="enableEmailBasedRecovery"
+                    name="enableEmailBasedRecovery"
+                    label={ t("extensions:manage.serverConfigurations.accountRecovery." +
+                        "usernameRecovery.form.fields.enableEmailBasedRecovery.label") }
+                    required={ false }
+                    readOnly={ readOnly }
+                    width={ 10 }
+                    data-testid={ `${ testId }-email-based-recovery` }
+                    data-componentid={ `${ testId }-email-based-recovery` }
+                />
+
+                {
+                    <Message info>
+                        <Icon name="info circle" />
+                        <Trans
+                            i18nKey={
+                                "extensions:manage.serverConfigurations.accountRecovery." +
+                                "passwordRecovery.form.smsProviderWarning"
+                            }
+                        >
+                            Ensure that an
+                            <Link
+                                external={ false }
+                                onClick={ () => {
+                                    history.push(
+                                        AppConstants.getPaths().get("SMS_PROVIDER")
+                                    );
+                                } }
+                            >SMS Provider
+                            </Link>
+                            &nbsp;is configured for the OTP feature to work properly.
+                        </Trans>
+                    </Message>
+                }
+
                 <Field.Checkbox
-                    className="mb-3"
-                    ariaLabel="notifyRecoverySuccess"
-                    name={ ServerConfigurationsConstants.USERNAME_RECOVERY_ENABLE }
-                    label="Enable username recovery"
+                    className="mt-3 mb-5"
+                    ariaLabel="enableSMSBasedRecovery"
+                    name="enableSMSBasedRecovery"
+                    label={ t("extensions:manage.serverConfigurations.accountRecovery." +
+                        "usernameRecovery.form.fields.enableSMSBasedRecovery.label") }
                     required={ false }
                     readOnly={ readOnly }
                     width={ 10 }
-                    data-testid={ `${testId}-notify-success` }
-                    toggle
-                    listen={ (value: boolean) => onSubmit(getUpdatedConfigurations(value)) }
-                    checked={ initialValue === "true" }
+                    data-testid={ `${ testId }-sms-based-recovery` }
```

Comment:
If there are E2E, this test id change can cause e2e to break.

Reply:
Removed the data-testid by https://github.com/wso2/identity-apps/pull/7004/commits/59d040047dd8b21efd511e781c167e8ebd65b64f